getReducedProblem.m

```diff
@@ -0,0 +1,144 @@
+function [newParameters,newProperties,newObjectiveFunction,newOptions] = getReducedProblem(parameters,properties,objectiveFunction,options)
+
+    % Definition of index set of optimized parameters
+    freePars = setdiff(1:parameters.number, options.fixedParameters);
+    fixedPars = options.fixedParameters;
+    fixedParValues = options.fixedParameterValues;
+   
+    %% Reduced parameter struct
+    % Basic information
+    newParameters.name = parameters.name(freePars);
+    newParameters.min = parameters.min(freePars);
+    newParameters.max = parameters.max(freePars);
+    newParameters.guess = parameters.guess(freePars);
+    newParameters.number = length(freePars);
+    if isfield(parameters, 'constraints')
```

Comment:
Please use some ANDs instead 3 levels of IFs and redundant assignment of defaults. Below: same